src/ol/disposable.js

```diff
@@ -0,0 +1,30 @@
+goog.provide('ol.Disposable');
+
+/**
+ * Objects that need to clean up after themselves.
+ * @constructor
+ */
+ol.Disposable = function() {};
+
+/**
+ * The object has already been disposed.
+ * @type {boolean}
+ * @private
+ */
+ol.Disposable.prototype.disposed_ = false;
```

Comment:
I thought it would confuse the compiler and prevent using `@struct` but I just tried and it compiled.
I see no major drawback in using this optimization.